src/extension/generalMobilePlatform.ts

```diff
@@ -81,6 +82,35 @@ export class GeneralMobilePlatform {
         return Promise.resolve();
     }
 
+    public modifyOptFromRunArgs(optName: string, value: any): void {
+        const isBinary = isBoolean(value);
+        const optIdx = this.runArguments.indexOf(optName);
+        if (optIdx > -1) {
+            if (isBinary && !value) {
+                this.runArguments.splice(optIdx, 1);
+            }
+            if (!isBinary) {
+                if (value === null && value === undefined) {
+                    this.runArguments.splice(optIdx, 2);
+                }
+                else {
+                    this.runArguments[optIdx + 1] = value;
+                }
+            }
+        }
+        else {
+            if (isBinary && value) {
+                this.runArguments.push(optName);
+            }
+            if (!isBinary) {
+                if (value !== null || value !== undefined) {
```

Comment:
If we make some changes on this function I suppose that this check will be not needed.